File: LeetCode/01-two-sum-1/solution.js
let twoSum = function (nums, target) {
  const nmap = new Map();
  for (let i = 0; i < nums.length; ++i) {
    const c = target - nums[i];
    if (nmap.has(c)) {
      return [nmap.get(c), i];
    }
    nmap.set(nums[i], i);
  }
};

let nlist = [2, 7, 11, 15];
let result = twoSum(nlist, 9);
console.log(`Results are: ${result}`);

File: LeetCode/01-two-sum-1/solution.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int,int> nmap;
    for(int i=0;i < nums.size();++i) {
        int c = target - nums[i];
        if(nmap.find(c) != nmap.end()) {
            return {nmap[c], i};
        }
        nmap[nums[i]] = i;
    }
    return {};
}

int main(int argc, char const *argv[])
{
    vector<int> nums = {2,7,11,15};
    auto v = twoSum(nums,9);
    cout << v[0] << " " << v[1];
    return 0;
}
